client.go

```diff
@@ -427,6 +430,20 @@ func (c *Client) stream(method, path string, streamOptions streamOptions) error
 		if err != nil {
 			return err
 		}
+
+		// ReadResponse may hang if server does not replay
+		connected := make(chan bool, 2)
```

Comment:
is connected intended to be used? i know this is the general pattern, just could use some explanation